sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_ulp_z3_wakeup.rs

```diff
@@ -0,0 +1,195 @@
+    Ok(())
+}
+
+fn wait_wakeup_time() -> () {
+    std::thread::sleep(Duration::from_micros(DEBOUNCE_SW_VALUE_USEC));
+}
+
+fn chip_sw_sysrst_ctrl_ulp_z3_wakeup(
+    opts: &Opts,
+    transport: &TransportWrapper,
+    config: &Config,
+    uart: &dyn Uart,
+    test_phase_addr: u32,
+    pins_initial: u8,
+    pins_wakeup: u8,
+) -> Result<()> {
+    // Reset target.
+    transport.reset_target(opts.init.bootstrap.options.reset_delay, true)?;
+    // Set pins to zero.
+    set_pins(transport, config, pins_initial)?;
+
+    // Tell device that we have driven pins to the initial value.
+    log::info!("Go to PhaseDriveInitial");
+    MemWriteReq::execute(&*uart, test_phase_addr, &[TestPhase::PhaseDriveInitial as u8])?;
+    // Wait for device to configure the wakeup source.
+    log::info!("Wait for device to configure wakeup source");
+    sync_with_sw(opts, uart)?;
+
+    // Wait for the debounce time: the pins are still in the initial value so it should not trigger.
+    log::info!("Wait debounce time.");
+    wait_wakeup_time();
+    // Make sure hardware block has not driven the wakeup pin.
+    log::info!("Make sure wakeup pin was not asserted");
+    ensure!(read_pins(transport, config)? == 0b0);
+    // Tell device that we have done nothing and it should check nothing happened.
+    log::info!("Go to PhaseWaitNoWakeup");
+    MemWriteReq::execute(&*uart, test_phase_addr, &[TestPhase::PhaseWaitNoWakeup as u8])?;
+    // Wait for device to go to sleep.
+    log::info!("Wait for device to sleep");
+    UartConsole::wait_for(uart, &TestStatus::InWfi.wait_pattern(), opts.timeout)?;
+
+    // We cannot simulate a glitch so simply trigger the wakeup.
+    set_pins(transport, config, pins_wakeup)?;
+
+    // Wait for device to wake up.
+    log::info!("Wait for device to wake up");
+    sync_with_sw(opts, uart)?;
+    // Tell device to check the wakeup cause.
+    log::info!("Go to PhaseWaitWakeup");
+    MemWriteReq::execute(&*uart, test_phase_addr, &[TestPhase::PhaseWaitWakeup as u8])?;
+    // Wait device to confirm.
+    log::info!("Wait for device to confirm wakeup cause");
+    sync_with_sw(opts, uart)?;
+    // Check that the hardware block is correctly driving the wakeup pin.
+    log::info!("Make sure wakeup pin was asserted");
+    ensure!(read_pins(transport, config)? == 0b1);
+
+    // Finish test.
+    log::info!("Go to PhaseDone");
+    MemWriteReq::execute(&*uart, test_phase_addr, &[TestPhase::PhaseDone as u8])?;
+
+    let _ = UartConsole::wait_for(&*uart, r"PASS!", opts.timeout)?;
+    Ok(())
+}
+
+fn main() -> Result<()> {
+    let opts = Opts::parse();
+    opts.init.init_logging();
+    let transport = opts.init.init_target()?;
+
+    /* Load the ELF binary and get the address of the `kPhase` variable
+     * in example_sival.c */
+    let elf_binary = fs::read(&opts.firmware_elf)?;
+    let elf_file = object::File::parse(&*elf_binary)?;
+    let mut symbols = HashMap::<String, (u32, u64)>::new();
```

Comment:
NIT: This is fine so don't bother changing it, but might be simpler to do something like this for just one symbol:

```rust
elf_file
    .symbols()
    .find(|symbol| symbol.name() == Ok("kTestPhaseReal"))
    .expect("Provided ELF missing 'kTestPhaseReal' symbol")
    .address()
```